app/controllers/taxa_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class TaxaController < ApplicationController
+  skip_before_action :login_required # there's no sensitive information here
+
+  def index
+    return render plain: 'Missing required parameter: term', status: :bad_request if params[:term].blank?
+
+    # Lookup by term and render results
+    term = params[:term]
+    taxon = client.taxon_from_text(term)
```

Comment:
question: should we call strip on the term to remove whitespace? The client makes "any-name/term" after url-encoding it, which would add %20 bits. The result would be an empty response.

note: if such a url is tested with browser, it would remove the trailing whitespace and hide this issue. Try it with a preceding space.

suggestion: the term might be enforced using params.require or an explicit check for presence.